File: internal/models/invoice.go
// Package models defines database-backed domain models.
package models

// InvoiceItem describes a single line item within an invoice.
type InvoiceItem struct {
	ID          int     `json:"id"`
	Description string  `json:"description"`
	Quantity    float64 `json:"quantity"`
	UnitPrice   float64 `json:"unitPrice"`
	Amount      float64 `json:"amount"`
}

// Invoice captures billing information for a client.
type Invoice struct {
	ID        int           `json:"id"`
	ClientID  int           `json:"clientId"`
	Number    string        `json:"number"`
	IssueDate string        `json:"issueDate"`
	DueDate   string        `json:"dueDate"`
	Subtotal  float64       `json:"subtotal"`
	TaxRate   float64       `json:"taxRate"`
	TaxAmount float64       `json:"taxAmount"`
	Total     float64       `json:"total"`
	Status    string        `json:"status"` // draft, sent, paid, overdue
	Items     []InvoiceItem `json:"items"`
}

File: frontend/src/schemas/invoice.ts
import { z } from "zod";
import { commonSchemas } from "./common";

export const invoiceItemSchema = z.object({
  id: z.number(), // Usually frontend temp id or backend id
  description: z.string().min(1),
  quantity: z.number().min(0),
  unitPrice: z.number().min(0),
  amount: z.number(),
});

export const invoiceSchema = z.object({
  id: z.optional(commonSchemas.id),
  clientId: z.number().int().positive(),
  number: commonSchemas.requiredString,
  issueDate: commonSchemas.dateString,
  dueDate: commonSchemas.dateString,
  items: z.array(invoiceItemSchema),
  subtotal: z.number(),
  taxRate: z.number().min(0).max(1),
  taxAmount: z.number(),
  total: z.number(),
  status: z.enum(["draft", "sent", "paid", "overdue"]),
});
